scripts/west_commands/runners/blackmagicprobe.py

```diff
@@ -51,16 +63,49 @@ def do_add_parser(cls, parser):
                             help='Assert SRST during connect? (default: no)')
 
     def bmp_flash(self, command, **kwargs):
-        if self.elf_file is None:
-            raise ValueError('Cannot debug; elf file is missing')
+
+        if self.file is not None:
+            # use file provided by the user
+            if not os.path.isfile(self.file):
+                err = 'Cannot flash; file ({}) not found'
+                raise ValueError(err.format(self.file))
+            if self.file_type in (FileType.HEX, FileType.ELF):
+                flash_cmd = f'"{self.file}"'
+            elif self.file_type == FileType.BIN:
+                if self.dt_flash:
+                    flash_addr = self.flash_address_from_build_conf(self.build_conf)
+                else:
+                    flash_addr = 0
+                flash_cmd = f'"{self.file}" 0x{flash_addr:x}'
+            else:
+                err = 'Cannot flash; jlink runner only supports hex and bin files'
+                raise ValueError(err)
+
+        else:
+            # use file provided by the buildsystem, preferring .hex over .bin over .elf
+            if self.hex_file is not None and os.path.isfile(self.hex_file):
+                flash_cmd = f'"{self.hex_file}"'
+            elif self.bin_file is not None and os.path.isfile(self.bin_file):
+                if self.dt_flash:
+                    flash_addr = self.flash_address_from_build_conf(self.build_conf)
+                else:
+                    flash_addr = 0
+                flash_cmd = f'"{self.bin_file}" 0x{flash_addr:x}'
+            elif self.elf_file is not None and os.path.isfile(self.elf_file):
+                flash_cmd = f'"{self.elf_file}"'
+            else:
+                raise ValueError('Cannot debug; no file to flash')
+
+        self.logger.info(f'Flashing file: {flash_cmd}')
```

Comment:
This doesn't only print the file when self.file_type is BIN and self.dt_flash is true; please fix to be consistent about just printing the file name in each case.